easybuild/easyblocks/g/gcc.py

```diff
@@ -380,7 +382,9 @@ def build_step(self):
                         raise EasyBuildError("Failed to change to %s: %s", libdir, err)
                     if lib == "gmp":
                         cmd = "./configure --prefix=%s " % stage2prefix
-                        cmd += "--with-pic --disable-shared --enable-cxx"
+                        cmd += "--with-pic --disable-shared --enable-cxx "
+                        if self.cfg['generic']:
```

Comment:
@raj76 @geimer Should we also consider the use of `--optarch=GENERIC` to enable this?

```python
from easybuild.tools.config import build_option
from easybuild.tools.toolchain.compiler import OPTARCH_GENERIC
...
            if build_option('optarch') == OPTARCH_GENERIC or self.cfg['generic']:
```

Having a dedicated easyconfig parameter makes sense, but maybe we should use `None` as default value.
That way we can allow people setting it to `False` to indicate that the generic build should *not* be done when `--optarch=GENERIC` is used. That's a bit of a weird use case, but I guess it's possible?

```python
if build_option('optarch') == OPTARCH_GENERIC and self.cfg['generic'] != False:
```

Reply:
I am OK with using  --optarch=GENERIC to enable this as well.

Reply:
I made the proposed changes in https://github.com/raj76/easybuild-easyblocks/pull/2, please review/merge to update this PR.

Reply:
I tested the updated `gcc.py` both with and without `--optarch=GENERIC`, works as intended.